### PR TITLE
Add custom roles with schema specific permission

### PR DESCRIPTION
**Pull request by akashkj**
To create custom role and its related permission, if it does not exist.

### CODE REVIEW

File: hq_superset/oauth.py [hunk 1]
```diff
@@ -10,3 +12,33 @@ def oauth_user_info(self, provider, response=None):
...
+        for domain_name in domains.items():
+            role_created = self.create_custom_role(domain_name)
+            if not role_created:
+                continue
+            self.create_schema(domain_name)
+            pv = self.create_custom_permission(domain_name)
+            sesh = self.get_session
+            role_created.permissions = pv
+            sesh.merge(role_created)
+            sesh.commit()
```
Comment (sravfeyn) [inline hq_superset/oauth.py:26]: What happens if there is an exception, should there be some sort of rollback?

File: hq_superset/oauth.py [hunk 2]
```diff
@@ -10,3 +12,33 @@ def oauth_user_info(self, provider, response=None):
...
+
+    def create_custom_permission(self, domain_name):
+        schema_access_perm_name = "schema_access"
+        schema_access_view_menu_name = "[{}].[{}]".format("superset", domain_name)
```
Comment (sravfeyn) [inline hq_superset/oauth.py:30]: What is this permission for?

**Comment by sravfeyn** [comment]
Looking good, Akash. I made few comments. I understand that these are the utils, but there is no code that's calling these yet?

**Comment by sravfeyn** [comment]
I ended up doing this myself here https://github.com/dimagi/hq_superset/pull/2/commits/e6b02aab2977d7d017ea83c0e0d366b11a3bf23f

**Comment by akashkj** [comment]
Is this PR not required now then? @sravfeyn

File: hq_superset/oauth.py (hunk 2 above)
Comment (akashkj) [reply hq_superset/oauth.py:30]: Format for schema access permission is <db>.<schema name>

**Comment by sravfeyn** [comment]
Yeah, I think we can close this

**Comment by akashkj** [comment]
Ref: https://github.com/dimagi/hq_superset/pull/7

**Pull request closed without merging**